src/Server.cpp

```diff
@@ -0,0 +1,211 @@
+#include "../inc/Server.hpp"
+
+/* ====== HELPER FUNCTIONS ====== */
+
+/**
+ * @brief Set the socket to nonblocking mode
+ * 
+ * Use fcntl to manipulate fd/sockets. Retrieve the currently set flags with
+ * F_GETFL. Set the flag with with F_SETFL. Use bitwise OR to set the 
+ * O_NONBLOCK bit in "flags" while preserving all other bits.
+ * @param sock	Socket to be manipulated
+ */
+static void setNonblocking(int sock)
+{
+	int	flags = fcntl(sock, F_GETFL, 0);
+	if (flags == -1) {
+		throw std::runtime_error("fcntl(F_GETFL)");
+	}
+	if (fcntl(sock, F_SETFL, flags | O_NONBLOCK) == -1) {
+		throw std::runtime_error("fcntl(F_SETFL)");
+	}
+}
+/* ====== CONSTRUCTOR/DESTRUCTOR ====== */
+
+/**
+ * @brief Constructor for the Server class.
+ * 
+ * This constructor initializes a server instance by creating a server socket,
+ * setting it to non-blocking mode, binding it to a specific address and port,
+ * listening on the server socket, creating an epoll instance for event handling,
+ * and adding the server socket to the epoll instance.
+ * 
+ * @throws std::runtime_error if any of the socket creation, binding, listening,
+ *         epoll creation, or epoll control operations fail.
+ * 
+ */
+Server::Server()
```

Comment:
Just to keep in mind: we will have one server, which depending on the config creates multiple virtual server (objects). So we will need to rename this I think (since it is one virtual server)
But: the main server needs to run() all virtual servers (we only are allowed to use one epoll)

Reply:
The code at present is handling the main server which then according to the incoming request will hand on the requests to the appropriate virtual servers. We could leave it as is and call the individual ones VirtualServer.